controllers/vrg_kubeobjects.go

```diff
@@ -0,0 +1,585 @@
+		request, err := KubeObjectsRecover(
+			v.ctx, v.reconciler.Client, v.reconciler.APIReader, v.log,
+			s3StoreAccessor.url,
+			s3StoreAccessor.bucketName,
+			s3StoreAccessor.regionName,
+			capturePathName,
+			s3StoreAccessor.veleroNamespaceSecretKeyRef,
+			sourceVrgNamespaceName, vrg.Namespace, recoverGroup.KubeObjectsSpec, veleroNamespaceName,
+			kubeObjectsCaptureName(captureNamePrefix, recoverGroup.BackupName, s3StoreAccessor.profileName),
+			kubeObjectsRecoverName(recoverNamePrefix, groupNumber), labels)
+		requests[groupNumber] = request
+
+		if err == nil {
+			v.log.Info("Kube objects group recovered", "number", capture.Number,
+				"group", groupNumber, "name", recoverGroup.BackupName, "profile", s3StoreAccessor.profileName,
+				"start", request.StartTime(), "end", request.EndTime())
+
+			continue
+		}
+
+		if errors.Is(err, KubeObjectsRequestProcessingError{}) {
+			v.log.Info("Kube objects group recovering", "number", capture.Number,
+				"group", groupNumber, "name", recoverGroup.BackupName, "profile", s3StoreAccessor.profileName,
+				"state", err.Error())
+
+			return err
+		}
+
+		v.log.Error(err, "Kube objects group recover error", "number", capture.Number,
+			"group", groupNumber, "name", recoverGroup.BackupName, "profile", s3StoreAccessor.profileName)
+
+		result.Requeue = true
+
+		return err
+	}
+
+	startTime := requests[0].StartTime()
+	duration := time.Since(startTime.Time)
+	v.log.Info("Kube objects recovered", "number", capture.Number, "profile", s3StoreAccessor.profileName,
+		"groups", len(groups), "start", startTime, "duration", duration)
+
+	return v.kubeObjectsRecoverRequestsDelete(result, veleroNamespaceName, labels)
+}
+
+func (v *VRGInstance) kubeObjectsRecoverRequestsDelete(
+	result *ctrl.Result, veleroNamespaceName string, labels map[string]string,
+) error {
+	if err := KubeObjectsRecoverRequestsDelete(v.ctx, v.reconciler.Client, veleroNamespaceName, labels); err != nil {
+		v.log.Error(err, "Kube objects recover requests delete error")
+
+		result.Requeue = true
+
+		return err
+	}
+
+	v.log.Info("Kube objects recover requests deleted")
+
+	return nil
+}
+
+func (v *VRGInstance) veleroNamespaceName() string {
+	veleroNamespaceName := VeleroNamespaceNameDefault
+
+	_, ramenConfig, err := ConfigMapGet(v.ctx, v.reconciler.APIReader)
+	if err != nil {
+		v.log.Error(err, "veleroNamespaceName config failed")
+
+		return veleroNamespaceName
+	}
+
+	if ramenConfig.KubeObjectProtection.VeleroNamespaceName != "" {
+		veleroNamespaceName = ramenConfig.KubeObjectProtection.VeleroNamespaceName
+	}
+
+	return veleroNamespaceName
+}
+
+func (v *VRGInstance) kubeObjectProtectionDisabled(caller string) bool {
+	vrgDisabled := v.instance.Spec.KubeObjectProtection == nil
+	cmDisabled := func() bool {
+		_, ramenConfig, err := ConfigMapGet(v.ctx, v.reconciler.APIReader)
+		if err != nil {
+			v.log.Error(err, "RamenConfig.KubeObjectProtection get failed")
+
+			return false
```

Comment:
Should probably return true here so that on failure, the kubeObject Protection would be disabled, until the error is resolved

Reply:
Fixed